### ISSUE TITLE
"Back Up Automatically" permissions borked.

### ISSUE THREAD

**Issue by nickavem** [flathub/org.gnome.DejaDup#6]
For some reason the first time to try and tick "Back up automatically" the program will freeze up and eventually crash. This will keep happening until you run `flatpak permission-reset org.gnome.DejaDup` then on the next launch it works perfectly fine. Have reproduced several times now by wiping the data and reinstalling  the flatpak.

OS: Fedora Silverblue 33 Beta
Version: 42.4

**Comment by mikix** [comment]
Ugh I saw that once but couldn't reproduce it! OK let me try again, using that permission-reset bit.

Thanks for the report.

**Comment by mikix** [comment]
OK, just played around and could not reproduce.  :(  I'll try installing silverblue next, maybe there is a specific issue there.

**Comment by mikix** [comment]
I can't reproduce on Silverblue 32 either. How are you wiping the data and reinstalling? I'm doing flatpak remove, flatpak permission-reset, and removing the ~/.var/app/org.gnome.DejaDup folder.

**Comment by KingAnde** [comment]
> I can't reproduce on Silverblue 32 either. How are you wiping the data and reinstalling? I'm doing flatpak remove, flatpak permission-reset, and removing the ~/.var/app/org.gnome.DejaDup folder.

The bug or the solution? Also, I was on SB33, although I doubt it would make nay difference?

My exact reproduction was on a VM with snapshots. I might be able to check a diff, but I'm not entirely sure.

**Comment by KingAnde** [comment]
Which permission in the manifest is for autostart? Now I'm thinking this could be upstream because I can't find anything on permissions besides `~/.var/app/org.gnome.DejaDup`, not even in `/etc` or `/var`.

**Comment by mikix** [comment]
It's not in the manifest, it's just a dbus request an app makes and a flatpak portal handles. Only way to control it is via `flatpak permission*` I think.

**Comment by Toadfield** [comment]
I have the same issue.
I'm on Manjaro Kde and installed it with the official repositories.
I use the newest stable version of Deja Dub.
It didn't work for me since newest kde update.

**Comment by nickavem** [comment]
> I have the same issue.
> I'm on Manjaro Kde and installed it with the official repositories.
> I use the newest stable version of Deja Dub.
> It didn't work for me since newest kde update.

The official `flathub` or `manjaro` repos? Have you tried `kbackup`?

**Comment by ahvth** [comment]
In the meantime, this issue can be solved on GNOME with 3 steps:

1. Install `flatseal` via flathub
2. In the `flatseal` GUI enable the D-Bus options (there are associated CLI commands for this as well, but I don't know them)
![Screenshot from 2020-11-27 01-03-17](https://user-images.githubusercontent.com/24368337/100397895-7431ca80-304c-11eb-8823-cc36b92a0ba1.png)
3. Enable "Run in background" in the GNOME Settings App"
![Screenshot from 2020-11-27 01-04-28](https://user-images.githubusercontent.com/24368337/100397920-93c8f300-304c-11eb-8f39-db22019b9fd6.png)

**Comment by nickavem** [comment]
Fix not working on F33 🙁️. I've just switched to native package 
for now as it keeps wanting to break itself.

**Comment by mikix** [comment]
If anyone can provide reproduction steps, I’d love to fix this. I haven’t had success myself making this happen intentionally.

**Comment by mikix** [comment]
OK, I believe I have fixed this in the latest update!

It was an interaction between my risky way of waiting for a flatpak portal response and a workaround I had for old flatpak portal versions (<1.5) that apparently causes problems on newer versions. The fix should work for both old and new versions.

Thanks all for helping me find this!

https://github.com/flathub/org.gnome.DejaDup/commit/9f2c08d4b10fe80d4cb63bad2a5c734189cde1d5

**Comment by nickavem** [comment]
Seems to be working for me! When turning it on initially, the app 
crashed, but then on another go it works so... victory-ish. 
🤷‍♂️️ I'll try and get some more useful logging info re: 
that crash.

Andrew J. Thurman
980-333-5159
